packages/cli/src/services/url.service.ts

```diff
@@ -0,0 +1,40 @@
+import { Service } from 'typedi';
+import config from '@/config';
+
+@Service()
+export class UrlService {
+	/** Returns the base URL n8n is reachable from */
+	readonly baseUrl: string;
+
+	constructor() {
+		this.baseUrl = this.generateBaseUrl();
+	}
+
+	/** Returns the base URL of the webhooks */
+	getWebhookBaseUrl() {
```

Comment:
Should we generate it only once and set it to a field, like `baseUrl`?

Reply:
we can't do that here just yet, because `process.env.WEBHOOK_URL` gets updated after tunnel setup finishes. This will be addressed in #6713